File: frontend/Ecommerce/src/app/review.ts
export interface Review {
    id?: number;
    productId: number;
    username: string;
    rating: number;
    comment: string;
    createdAt?: string;
}

export interface ReviewDTO {
    id: number;
    comment: string;
    rating: number;
    timestamp: string;
    userName: string;
    userImage: string;
    productName: string;
    mediaList: ReviewMediaDTO[];
  }
  
  export interface ReviewMediaDTO {
    id: number;
    type: string;
    url: string;
  }
  

File: backend/Ecommerce/src/main/java/com/Ojt/Ecommerce/entity/Review.java
package com.Ojt.Ecommerce.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "review")
@Getter
@Setter
public class Review {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "product_id")
    private Product product;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id")
    private User user;

    @Column(columnDefinition = "TEXT")
    private String comment;

    private int rating;

    private LocalDateTime timestamp;

    @OneToMany(mappedBy = "review", cascade = CascadeType.ALL, fetch = FetchType.EAGER, orphanRemoval = true)
    private List<ReviewMedia> mediaList = new ArrayList<>();

    @PrePersist
    protected void onCreate() {
        timestamp = LocalDateTime.now();
    }
}
